doc/patch/patch-CI-IO.patch

```diff
@@ -0,0 +1,150 @@
+From 23e6b0da8dbdf3f0c6fb5e962d3bc753eb6d078d Mon Sep 17 00:00:00 2001
+From: Tim Griesbach <tim.griesbach@uni-bonn.de>
+Date: Wed, 15 Jan 2025 15:23:29 +0100
+Subject: [PATCH] Implement sleep and retry in case of busy file system
+
+---
+ src/sc_io.c | 92 +++++++++++++++++++++++++++++++++++++++++++++++++++++
+ 1 file changed, 92 insertions(+)
+
+diff --git a/src/sc_io.c b/src/sc_io.c
+index 4cfc7b81..fb3f065c 100644
+--- a/src/sc_io.c
++++ b/src/sc_io.c
+@@ -24,6 +24,13 @@
+ #include <sc_io.h>
+ #include <sc_puff.h>
+ #include <libb64.h>
++#ifdef _MSC_VER
++#include <windows.h>
++#elif _POSIX_C_SOURCE >= 199309L
++#include <time.h>
++#else
++#include <unistd.h>
++#endif
+ #ifdef SC_HAVE_ZLIB
+ #include <zlib.h>
+ #else
+@@ -37,6 +44,60 @@
+ #include <errno.h>
+ #endif
+ 
++/** In case of an unexpected retrieved count 0, this macro sleeps and retries.
++ * In the Github Actions CI, we experienced transient CI failures due to the
++ * file system not performing requested I/O operations.
++ *
++ * The macro can be only callled in the case that expected count is unequal 0.
++ * It checks if count is 0 and if this is true, it sleeps and then retries the
++ * I/O operation once.
++ */
++#define SC_IO_SLEEP_AND_RETRY(func, time) do {\
++                                    if (*ocount == 0) {\
++                                      sc_io_sleep (time);\
++                                      mpiret = func (mpifile, offset,\
++                                                     (void *) ptr, count, t,\
++                                                     &mpistatus);\
++                                      if (mpiret == sc_MPI_SUCCESS) {\
++                                        mpiret = sc_MPI_Get_count (&mpistatus,\
++                                                                   t, ocount);\
++                                        SC_CHECK_MPI (mpiret);\
++                                        return sc_MPI_SUCCESS;\
++                                      }\
++                                      retval = sc_io_error_class (mpiret,\
++                                                                  &errcode);\
++                                      SC_CHECK_MPI (retval);\
++                                      return errcode;\
++                                    }} while (0)
++
++
++/* This function is only needed if MPI IO is available */
++#ifdef SC_ENABLE_MPIIO
```

Comment:
Is this kind of portability code replicated elsewhere in the code base? We may instead provide a helper function for this kind of sleep that is always available to all of libsc and libraries using it.

Reply:
I moved the portable sleep function as `sc_sleep` to `sc.{c,h}`.

Reply:
Cool!